recipes/gdcm/all/conanfile.py

```diff
@@ -0,0 +1,228 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+class GDCMConan(ConanFile):
+    name = "gdcm"
+    topics = ("dicom", "images")
+    homepage = "http://gdcm.sourceforge.net/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    description = "C++ library for DICOM medical files"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if (self.settings.compiler == "Visual Studio"
+                and self.options.shared
+                and str(self.settings.compiler.runtime).startswith("MT")):
+            raise ConanInvalidConfiguration("shared gdcm can't be built with MT or MTd")
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "11")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["GDCM_BUILD_DOCBOOK_MANPAGES"] = "OFF"
+        self._cmake.definitions["GDCM_BUILD_SHARED_LIBS"] = "ON" if self.options.shared else "OFF"
+        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = "ON"
```

Comment:
conan handles the translation

```suggestion
        self._cmake.definitions["GDCM_BUILD_DOCBOOK_MANPAGES"] = False
        self._cmake.definitions["GDCM_BUILD_SHARED_LIBS"] = self.options.shared
        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = True
```

Reply:
conan doesn't transform True/False to "ON"/"OFF"

```
//Build man pages from XML docbook ?
GDCM_BUILD_DOCBOOK_MANPAGES:BOOL=False
```

Reply:
That's what we want? There's no `STREQL` in the upstream cmake for 'ON'

Reply:
I'm personally not used to having True/False in CMake. It's more common to have ON/OFF and I've even seen strange cases where people check for these specific values. But if you've already analyzed upstream, then ok :)